open-commerce-search-api/src/main/java/de/cxp/ocs/model/query/Query.java

```diff
@@ -0,0 +1,118 @@
+package de.cxp.ocs.model.query;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+/**
+ * Represent a query as a string, parse a string into a Query.
+ * 
+ * This implementation works on & delimited key/value pairs, handling each pair as a refinement i.e. attribute/value.
+ * The special _s attribute denotes a search phrase
+ * 
+ * NB this implementation is incomplete
+ * 
+ * @author pavel
+ *
+ */
+@Data
+@Accessors(chain = true)
+public class Query {
+
+	private String searchPhrase;
+	private List<Refinement> refinements = new ArrayList<Query.Refinement>();
+	private final Style style;
+	
+	// Annoyingly, Jax-rs or Jersey automatically resolve $ as sort of a template parameter
+	// thus $s=dress is not possible out of the box.
+	private static final String SEARCH = "_s";
+	
+	public static enum Style { 
+		DSL("/"), URL("&");
+
+		private final String separator;
+
+		Style(String separator) {
+			this.separator = separator;
+		}
+		
+		public String getSeparator() {
+			return separator;
+		}
+	};
+	
+	public Query(String query, Style style) {
+		this.style = style;
+	    String[] pairs = query.split(style.separator);
+	    
+	    for (String pair : pairs) {
+	        int index = pair.indexOf("=");
+	        try {
+				String attribute = URLDecoder.decode(pair.substring(0, index), "UTF-8");
+				String values = URLDecoder.decode(pair.substring(index + 1), "UTF-8");
+				
+				if (attribute.equals(SEARCH)) {
+					searchPhrase = values;
+				} else {
+					refinements.add(new Refinement(attribute, values.split(",")));
+				}
```

Comment:
I thought about it a bit and I think its not a good idea to put such utility logic into the API. This is standard URL parsing here with some special handling to transform it into that "rich" Query model. But it has no advantage since the service has to parse and validate that again.
Additionately such utility stuff can't be used by non-Java clients.